### PR TITLE
Better test for cache method availability

### LINKED ISSUE #54737: Constructed inventory plugin fails due to 'update_cache_if_changed'
##### SUMMARY

The `constructed` plugin fails to parse previously parseable yml files.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

constructed

##### ANSIBLE VERSION

```paste below
ansible 2.8.0.dev0
  config file = None
  configured module search path = ['/home/nikos/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/nikos/Projects/ansible/lib/ansible
  executable location = /home/nikos/Projects/ansible/bin/ansible
  python version = 3.7.2 (default, Jan 10 2019, 23:51:51) [GCC 8.2.1 20181127]
```

##### CONFIGURATION

```paste below
ANSIBLE_FORCE_COLOR(/home/nikos/Projects/one/ansible/ansible.cfg) = True
ANSIBLE_NOCOWS(/home/nikos/Projects/one/ansible/ansible.cfg) = False
ANSIBLE_PIPELINING(/home/nikos/Projects/one/ansible/ansible.cfg) = True
ANSIBLE_SSH_ARGS(/home/nikos/Projects/one/ansible/ansible.cfg) = -o ControlPath=/var/shm/control:%h:%p:%r
ANY_ERRORS_FATAL(/home/nikos/Projects/one/ansible/ansible.cfg) = True
CACHE_PLUGIN(/home/nikos/Projects/one/ansible/ansible.cfg) = yaml
CACHE_PLUGIN_CONNECTION(/home/nikos/Projects/one/ansible/ansible.cfg) = .tmp/facts_cache
CACHE_PLUGIN_TIMEOUT(/home/nikos/Projects/one/ansible/ansible.cfg) = 7200
COLOR_DEBUG(/home/nikos/Projects/one/ansible/ansible.cfg) = magenta
DEFAULT_CALLBACK_PLUGIN_PATH(/home/nikos/Projects/one/ansible/ansible.cfg) = ['/home/nikos/Projects/one/ansible/lib/plugins/callback']
DEFAULT_EXECUTABLE(/home/nikos/Projects/one/ansible/ansible.cfg) = /bin/bash
DEFAULT_FILTER_PLUGIN_PATH(/home/nikos/Projects/one/ansible/ansible.cfg) = ['/home/nikos/Projects/one/ansible/lib/plugins/filter']
DEFAULT_GATHERING(/home/nikos/Projects/one/ansible/ansible.cfg) = smart
DEFAULT_HOST_LIST(/home/nikos/Projects/one/ansible/ansible.cfg) = ['/home/nikos/Projects/one/vars/active/inventory']
DEFAULT_INVENTORY_PLUGIN_PATH(/home/nikos/Projects/one/ansible/ansible.cfg) = ['/home/nikos/Projects/one/ansible/lib/plugins/inventory']
DEFAULT_LOOKUP_PLUGIN_PATH(/home/nikos/Projects/one/ansible/ansible.cfg) = ['/home/nikos/Projects/one/ansible/lib/plugins/lookup']
DEFAULT_MANAGED_STR(/home/nikos/Projects/one/ansible/ansible.cfg) = Ansible managed (do not edit, changes may be overwritten)
DEFAULT_MODULE_PATH(/home/nikos/Projects/one/ansible/ansible.cfg) = ['/home/nikos/Projects/one/ansible/lib/modules']
DEFAULT_MODULE_UTILS_PATH(/home/nikos/Projects/one/ansible/ansible.cfg) = ['/home/nikos/Projects/one/ansible/lib/module_utils']
DEFAULT_PRIVATE_KEY_FILE(/home/nikos/Projects/one/ansible/ansible.cfg) = /home/nikos/Projects/one/gce/ssh_key
DEFAULT_ROLES_PATH(/home/nikos/Projects/one/ansible/ansible.cfg) = ['/home/nikos/Projects/one/roles', '/home/nikos/Projects/one/role>
DEFAULT_STDOUT_CALLBACK(/home/nikos/Projects/one/ansible/ansible.cfg) = anstomlog
DEFAULT_STRATEGY(/home/nikos/Projects/one/ansible/ansible.cfg) = linear
DEFAULT_TIMEOUT(/home/nikos/Projects/one/ansible/ansible.cfg) = 30
DEFAULT_TRANSPORT(/home/nikos/Projects/one/ansible/ansible.cfg) = ssh
DEFAULT_UNDEFINED_VAR_BEHAVIOR(/home/nikos/Projects/one/ansible/ansible.cfg) = True
DEFAULT_VARS_PLUGIN_PATH(/home/nikos/Projects/one/ansible/ansible.cfg) = ['/home/nikos/Projects/one/ansible/lib/plugins/vars']
DEPRECATION_WARNINGS(/home/nikos/Projects/one/ansible/ansible.cfg) = False
DISPLAY_ARGS_TO_STDOUT(/home/nikos/Projects/one/ansible/ansible.cfg) = True
DISPLAY_SKIPPED_HOSTS(/home/nikos/Projects/one/ansible/ansible.cfg) = False
HOST_KEY_CHECKING(/home/nikos/Projects/one/ansible/ansible.cfg) = False
INVALID_TASK_ATTRIBUTE_FAILED(/home/nikos/Projects/one/ansible/ansible.cfg) = True
INVENTORY_ENABLED(/home/nikos/Projects/one/ansible/ansible.cfg) = ['custom', 'scenario', 'constructed', 'yaml']
RETRY_FILES_ENABLED(/home/nikos/Projects/one/ansible/ansible.cfg) = False
USE_PERSISTENT_CONNECTIONS(/home/nikos/Projects/one/ansible/ansible.cfg) = True
```

##### OS / ENVIRONMENT

Arch linux latest

##### STEPS TO REPRODUCE

I have the following config file:

`inventory.config`:
```
---
# vi: ft=yaml

plugin: constructed
strict: true
groups:
  _network: true
  _passwords: true
keyed_groups:
  - prefix: ''
    key: inventory_hostname | unnamespace
```

##### EXPECTED RESULTS

I would expect the inventory to be parsed.

##### ACTUAL RESULTS

```paste below
yaml declined parsing /home/nikos/Projects/one/vars/active/inventory/50-inventory.config as it did not pass it's verify_file() method
 [WARNING]:  * Failed to parse /home/nikos/Projects/one/vars/active/inventory/50-inventory.config with custom plugin: Incorrect plugin name in file: constructed

  File "/home/nikos/Projects/ansible/lib/ansible/inventory/manager.py", line 272, in parse_source
    plugin.parse(self._inventory, self._loader, source, cache=cache)
  File "/home/nikos/Projects/one/ansible/lib/plugins/inventory/custom.py", line 63, in parse
    self._read_config_data(path)
  File "/home/nikos/Projects/ansible/lib/ansible/plugins/inventory/__init__.py", line 224, in _read_config_data
    raise AnsibleParserError("Incorrect plugin name in file: %s" % config.get('plugin', 'none found'))

 [WARNING]:  * Failed to parse /home/nikos/Projects/one/vars/active/inventory/50-inventory.config with scenario plugin: Incorrect plugin name in file: constructed

  File "/home/nikos/Projects/ansible/lib/ansible/inventory/manager.py", line 272, in parse_source
    plugin.parse(self._inventory, self._loader, source, cache=cache)
  File "/home/nikos/Projects/one/ansible/lib/plugins/inventory/scenario.py", line 61, in parse
    self._read_config_data(path)
  File "/home/nikos/Projects/ansible/lib/ansible/plugins/inventory/__init__.py", line 224, in _read_config_data
    raise AnsibleParserError("Incorrect plugin name in file: %s" % config.get('plugin', 'none found'))

 [WARNING]:  * Failed to parse /home/nikos/Projects/one/vars/active/inventory/50-inventory.config with constructed plugin: 'InventoryModule' object has no attribute
'update_cache_if_changed'

  File "/home/nikos/Projects/ansible/lib/ansible/inventory/manager.py", line 274, in parse_source
    plugin.update_cache_if_changed()

 [WARNING]: Unable to parse /home/nikos/Projects/one/vars/active/inventory/50-inventory.config as an inventory source

```

### PR DESCRIPTION
**Pull request by bcoca**
fixes #54737

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

inventory

### CODE REVIEW

**Review by s-hertel — APPROVED** [review]
This would be good to fix in auto too, if plugin._cache is not predictably created via Cacheable.

File: changelogs/fragments/fix_cache_assumption.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+    - use class inheritance vs internal attribute indicator before executing inherited method for inventory plugins.
```
Comment (nitzmahone) [inline changelogs/fragments/fix_cache_assumption.yml]: no longer accurate, now "duck typed" ;)

**Review by nitzmahone — CHANGES_REQUESTED** [review]
just changelog is out of date (maybe might want to also include a bugfix line about tweaking the instancing behavior)

**Review by s-hertel — COMMENTED** [review]
Looks good besides changelog

**Review by nitzmahone — APPROVED** [review]

**Pull request merged**